### DISCUSSION TITLE
how to get isinstance() to pass?

### DISCUSSION THREAD

**Discussion by LegNeato** [PyO3/pyo3#1970] [Questions] [answered]
I want my Rust code to pass a python `isinstance()` check for the standard python Uuid class. Is this currently supported?

(I think I might have asked this in chat months ago but can't find the answer)

**Comment by birkenfeld** [comment]
Tricky: the magic method `__instancecheck__` exists to allow this, but it must be defined in the metaclass. I don't think we do anything with metaclasses in PyO3 so far.

This is why `isinstance` is not idiomatic Python in most cases. It quacks like a duck etc.

**Reply by LegNeato** [reply]
Yeah, I guess I can patch it in Django but it will take time for it to release and for us to upgrade :-(

**Comment by birkenfeld** [comment, marked as answer]
Creating types with custom metaclass seems to be hard to impossible (for abi3) right now from the C API. See https://bugs.python.org/issue15870 for a discussion.

**Comment by LegNeato** [comment]
Looks like there is a task tracking: https://github.com/PyO3/pyo3/issues/906
